net_transport.go

```diff
@@ -393,6 +406,14 @@ func (n *NetworkTransport) InstallSnapshot(id ServerID, target ServerAddress, ar
 		return err
 	}
 
+	if halfClosableConn, ok := conn.conn.(HalfClosableConn); ok {
+		// Close the writing half of the connection to indicate all data is sent
+		err = halfClosableConn.CloseWrite()
```

Comment:
I'll note that we've had problems with Consul snapshots inside a docker container because it doesn't properly support half close https://github.com/docker/libnetwork/issues/1204. Unclear if newer versions of docker have fixed this. @slackpad any thoughts on whether this is an acceptable risk ? ie. merging this may increase the blast radius of potential issues with using snapshots inside Consul/Nomad when they are running in containers.

Reply:
That's a good point - I'd make the use of half-close opt-in so people can consciously use it. For Consul and Nomad I think we'd leave as-is (the API-driven snapshot and restore operations use the half-close for their internal RPCs and we have seen folks hit issues there).

Reply:
I could add a check to only do this when `Size == -1`. Would that be sufficient?

Reply:
That seems to unnecessarily couple things. I see half close as a good thing to do, we just need to make it opt in, so having a boolean for it in the request, and defaulting to false would work so we can disable it.